crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -61,13 +62,17 @@ pub(crate) async fn fetch_owned_objs(
     _filter: Option<ObjectFilter>,
 ) -> Result<Connection<String, Object>> {
     if before.is_some() && after.is_some() {
-        return Err(Error::new("before and after must not be used together"));
+        return Err(
+            CustomError::Input("before and after must not be used together".to_string()).extend(),
```

Comment:
A primary benefit of using `thiserror` is that you don't need to provide the error copy inline, when you're creating the error -- you only need to provide any additional context.  I would recommend that you create specific cases for these errors:

```rust
enum CustomError {
    #[error("'before' and 'after' must not be used together.")]
    CursorNoBeforeAfter,

    #[error("'first' and 'last' must not be used together.")]
    CursorNoFirstLast,
    
    #[error("Reverse pagination is not supported.")]
    CursorNoReversePagination,

    // ... etc
}
```